src/spa_solver_visualizer.cpp

```diff
@@ -0,0 +1,103 @@
+#include "spa_solver_visualizer.h"
+#include "ros/console.h"
+#include <pluginlib/class_list_macros.h>
+
+namespace karto_plugins{
+
+SPASolverVisualizer::SPASolverVisualizer():marker_count_(0),visualizer_initialized_(false)
+{
+}
+
+void SPASolverVisualizer::initialize(const boost::shared_ptr<karto::ScanSolver>& solver){
```

Comment:
I think the dynamic cast is fine in this case. It is not called in a loop.
See also http://stackoverflow.com/questions/580270/avoiding-dynamic-cast-rtti.